### PR TITLE
Added SSE client for CLI to wait for operation to end

### PR DESCRIPTION
**Pull request by dhruv0000**
**Description**

This adds a SSE client for CLI which it can use to wait for the operation to be over or end in case of timeout (20 mins). 
Need to use `watch` flag for it to work.

**Notes for Reviewers**
- SMI Github Action currently uses a timeout as a guess to check if the test is complete. This PR will address that issue too.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Task linked: [CU-4hbp5y Event client Handler](https://app.clickup.com/t/4hbp5y)

**Comment by DelusionalOptimist** [comment]
```
[  7:45PM ]  [ azureuser@test-vm:~/dev/meshery/mesheryctl(master-dhr✗) ]
 $ ./mesheryctl mesh validate --spec smi --adapter meshery-osm:10009 -t ~/Downloads/auth.json --watch
A new release of mesheryctl is available: Not Set → v0.5.41
https://github.com/layer5io/meshery/releases/tag/v0.5.41
Check https://docs.meshery.io/guides/upgrade#upgrading-meshery-cli for instructions on how to update mesheryctl

Starting service mesh validation...
{}
Verifying Operation
⣾ Validation started
⣽ Validation started
⣻ Validation started
⢿ Validation started
 Event :map[operation_id:ecbfac49-6c5a-4e8d-b7ac-f82573eab346 summary:SMI Conformance test completed successfully]

Validation complete
[  7:45PM ]  [ azureuser@test-vm:~/dev/meshery/mesheryctl(master-dhr✗) ]
 $
```
@dhruv0000 Apparently this still doesn't wait for the SMI conformance operation to complete : /

**Comment by dhruv0000** [comment]
> SMI Conformance test completed successfully

@DelusionalOptimist I am a bit confused. Isnt `Event :map[operation_id:ecbfac49-6c5a-4e8d-b7ac-f82573eab346 summary:SMI Conformance test completed successfully]
`  the event we are looking for, and it looks like it ended after it saw the event.

Am I missing something?

File: mesheryctl/internal/cli/root/mesh/validate.go [hunk 1]
```diff
@@ -73,6 +76,9 @@ var validateCmd = &cobra.Command{
 			return err
 		}
 
+		s := utils.CreateDefaultSpinner("Validation started", "\nValidation complete")
+
+		s.Start()
```
Comment (DelusionalOptimist) [inline mesheryctl/internal/cli/root/mesh/validate.go:81]:
```suggestion
```

File: mesheryctl/internal/cli/root/mesh/validate.go [hunk 2]
```diff
@@ -133,6 +139,17 @@ var validateCmd = &cobra.Command{
 		}
 
 		log.Infof(string(body))
+		s.Stop()
```
Comment (DelusionalOptimist) [inline mesheryctl/internal/cli/root/mesh/validate.go:142]:
```suggestion

```

**Review by DelusionalOptimist — CHANGES_REQUESTED** [review]

**Review by DelusionalOptimist — APPROVED** [review]
LGTM & CITY

**Review by DelusionalOptimist — APPROVED** [review]
LGTM

**Pull request merged**